src/web/web-service/wpan_service.cpp

```diff
@@ -66,20 +70,37 @@ std::string WpanService::HandleJoinNetworkRequest(const std::string &aJoinReques
     VerifyOrExit(client.Connect(), ret = kWpanStatus_SetFailed);
 
     VerifyOrExit(reader.parse(aJoinRequest.c_str(), root) == true, ret = kWpanStatus_ParseRequestFailed);
-    index        = root["index"].asUInt();
-    masterKey    = root["masterKey"].asString();
-    prefix       = root["prefix"].asString();
-    defaultRoute = root["defaultRoute"].asBool();
+    index          = root["index"].asUInt();
+    credentialType = root["credentialType"].asString();
+    masterKey      = root["masterKey"].asString();
+    pskd           = root["pskd"].asString();
```

Comment:
```
if (credentialType == CREDENTIAL_TYPE_MASTER_KEY)
{
    masterKey = root["masterKey"].asString();
}
else if (credentialType == CREDENTIAL_TYPE_PSKD)
{
    pskd = root["pskd"].asString();
}
    
```
Not a MUST now. Probably we can enhance the UI later to update the table according user's choice of `credentialType`.

Reply:
Thanks. Will consider this way in later versions when I add more credential types.